From f419f56ec8715e980aeb673bf4afd7649580d53c Mon Sep 17 00:00:00 2001
From: Jesse Barnes <jbarnes@nietzche.virtuousgeek.org>
Date: Mon, 12 May 2008 10:58:24 -0700
Subject: [PATCH] Panel fitting:  fix letterbox modes

In full_aspect mode, we try to preserve the aspect ratio by adding
either top & bottom or left & right borders.  In the letterbox case (top
& bottom borders) we were miscalculating the top border which led to
programming a bad mode.  Fix the calculation and bug #15559.
(cherry picked from commit 5103e5a39a4869a714b8a59da8bf51ecc8c65e4f)

diff --git a/src/i830_lvds.c b/src/i830_lvds.c
index 48402df..899c6cb 100644
--- a/src/i830_lvds.c
+++ b/src/i830_lvds.c
@@ -671,7 +671,7 @@ i830_lvds_mode_fixup(xf86OutputPtr output, DisplayModePtr mode,
 
 		/* Letterbox will have top/bottom borders */
 		top_border = (dev_priv->panel_fixed_mode->VDisplay -
-			      mode->VDisplay) / 2;
+			      scaled_height) / 2;
 		bottom_border = top_border;
 		if (mode->VDisplay & 1)
 		    bottom_border++;
-- 
1.5.5.1

